python/tank/util/filesystem.py

```diff
@@ -355,64 +355,38 @@ def get_permissions(path):
     """
     return stat.S_IMODE(os.lstat(path)[stat.ST_MODE])
 
-@with_cleared_umask
-def add_permissions(path, permissions=0775):
-    """
-    Add the permissions to a file system item at the path given
-
-    :param path: Path to the file or folder whose permissions will be added to
-    :param permissions: permissions to add
-    :raises: OSError - if there was a problem retrieving permissions for the path
-    """
-    os.chmod(path, get_permissions(path) | permissions)
-
-
-def add_permissions_recursive(path, permissions=0775):
-    """
-    Add the permissions to a folder in path and to all of its 
-    contents recursively
-    
-    :param path: Path to the folder whose permissions will be added to
-    :param permissions: permissions to add    
-    :returns: True if all files and folders new permissions could be set,
-              False otherwise. Files that can't be changed will be logged
-              as errors and skipped
-    """
-    noError = True
-    for root, dirs, files in os.walk(path, topdown=False):
-        for folder in [os.path.join(root, d) for d in dirs]:
-            try:
-                add_permissions(folder, permissions)
-            except Exception, e:
-                log.error("Could not change permissions on %s: %s" % (folder, e))
-                noError = False
-        for file in [os.path.join(root, f) for f in files]:
-            try:
-                add_permissions(file, permissions)
-            except Exception, e:
-                log.error("Could not change permissions on %s: %s" % (file, e))
-                noError = False
-    return noError
-
 def delete_folder(path):
     """
     Deletes a folder and all of its contents recursively
 
     :param path: File system path to location to the folder to be deleted
-    :returns: True if the folder was deleted, False otherwise
+    :returns: True if the folder and all of its contents were deleted without
+              any contingencies. False if there were any issues removing
+              any items in the folder. Problems deleting any items will be
+              reported as errors in the log output but otherwise ignored and
+              skipped; meaning the function will continue deleting as much
+              as it can.
     """
+    deleted = []
```

Comment:
You had me stumped for a bit, I didn't understand what you were trying to do here. If I understand correctly, you are trying to set some variable outside of your scope.
This can be done in a cleaner fashion.

```
deleted = True

def _on_rm_error(func, path, exc_info):
    ...
   try:
       ...
   except:
       global deleted
       deleted = True
```

What the global keyword does is it tells Python that any reference to the deleted variable in the current scope (in this case, the `_on_rm_error method`) actually refers to a variable in an enclosing scope (`delete_folder` in this case)